src/store/entities/selectors.ts

```diff
@@ -0,0 +1,71 @@
+import set from 'lodash.set';
+import get from 'lodash.get';
+
+import decodeQueryCacheKey from '../../utils/decodeQueryCacheKey';
+
+import { State } from '../reducer';
+import { State as EntitiesState } from '../entities/reducer';
+import { Entity } from '../../typings/API';
+
+/**
+ * @TODO
+ * Helper function to determine how to convert an api entity reference; f.e.
+ * { type: 'contact', id: 'test123' }
+ * into the domain type "contacts", should this be revised in the api or should we
+ * change our store to store directly by type, or create a mapping?
+ */
+export const convertAPIEntityTypeToDomain = (type: string) => {
+  return `${type}s`;
+};
+
+/**
+ * Helper function to merge entities into their respective paths
+ * Returns a new, non-mutated entity object
+ */
+export const mergeEntitiesIntoPaths = (entities: EntitiesState, paths: string[], entity: Entity) => {
+  const clonedEntity = { ...entity };
```

Comment:
Q: Will `clonedEntity = entity` just create a reference?

Reply:
Yep, it does, and it's mostly because `set` mutates the object

Reply:
Be aware that `const clonedEntity = { ...entity };` only does a shallow clone, so only the first level of that object will be new instances, deeper levels will still be references.

Reply:
Fair enough, I looked some stuff up and tried some stuff out after your comment @lowiebenoot 	. So to be really immutable here, shouldn't we recursively clone the entity? Or will it never be deeply nested?

Reply:
@lowiebenoot Aha, wasn't aware of that, and it's actually possible to set a sideload reference at a deeper path, so I should adjust it, I'm going to try to use `immer` instead to produce the merged entity.

Reply:
good sesh guys 💃